### PR TITLE
tree: Add tree FAQ

### PR DESCRIPTION
**Pull request by CraigMacomber**
## Description

Add FAQ to tree.

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

### CODE REVIEW

**Review by noencke — APPROVED** [review]

File: packages/dds/tree/README.md [hunk 1]
```diff
@@ -129,6 +129,35 @@ This package can be developed using any of the [regular workflows for working on
     both of these are provided by the mocha testing extension thats recommended by the workspace.
     Note that this does not build the tests, so always be sure to build first.
 
+## Frequently asked questions:
+
+### Why can't I assign insertable content to a field?
+``` typescript
```
Comment (Josmithr) [inline packages/dds/tree/README.md:135]: Nit (Markdown spec calls for a line break after a heading)
```suggestion

``` typescript
```

File: packages/dds/tree/README.md [hunk 2]
```diff
@@ -129,6 +129,35 @@ This package can be developed using any of the [regular workflows for working on
...
+
+const factory = new SchemaFactory("com.fluidframework.faq");
+class Empty extends factory.object("Empty", {}) {}
+class Test extends factory.object("Test", { data: Empty }) {}
+function set(node: Test) {
+	node.data = {}; // Why does this not compile?
+}
+```
+
+This is due to [a limitation of the TypeScript language](https://github.com/microsoft/TypeScript/issues/43826) which makes it impossible for tree to allow that to type check while keeping the strong typing on the getters for reading data.
```
Comment (Josmithr) [inline packages/dds/tree/README.md:146]: Nit
```suggestion
This is due to [a limitation of the TypeScript language](https://github.com/microsoft/TypeScript/issues/43826) which makes it impossible for tree to allow that to type-check while keeping the strong typing on the getters for reading data.
```

File: packages/dds/tree/README.md [hunk 3]
```diff
@@ -129,6 +129,35 @@ This package can be developed using any of the [regular workflows for working on
...
+
+To workaround this, create an unhydrated node:
+
+``` typescript
+node.data = new Empty({}); // The unhydrated node's type matches the type returned by the getter, and thus is compatible with the setter
+```
+
+Insertable content can still be used in other places, like when nested in other insertable content, in ArrayNode editing methods and initializing views.
```
Comment (Josmithr) [inline packages/dds/tree/README.md:154]: Nit
```suggestion
Insertable content can still be used in other places, like when nested in other insertable content, in ArrayNode editing methods, and when initializing views.
```

File: packages/dds/tree/README.md [hunk 4]
```diff
@@ -129,6 +129,35 @@ This package can be developed using any of the [regular workflows for working on
     both of these are provided by the mocha testing extension thats recommended by the workspace.
     Note that this does not build the tests, so always be sure to build first.
 
+## Frequently asked questions:
```
Comment (Josmithr) [inline packages/dds/tree/README.md:132]: Already a heading, so doesn't really need a `:`

```suggestion
## Frequently asked questions
```

**Review by Josmithr — APPROVED** [review]

**Pull request merged**